utils/data-processing-task-submitter-container/src/test/java/com/github/cafdataprocessing/utilities/tasksubmitter/MainIT.java

```diff
@@ -105,21 +108,32 @@ private void checkTaskMessage(TaskMessage message, Collection<String> encountere
-        Assert.assertEquals(taskData.getProjectId(), projectId, "Project ID on task should match that specified in properties.");
-        Assert.assertEquals(taskData.getWorkflowId(), Long.toString(workflowId), "Workflow ID on task should match that specified in properties.");
-        Assert.assertEquals(taskData.getOutputPartialReference(), expectedDataDir,
+        DocumentWorkerDocumentTask taskData = codec.deserialise(taskDataBytes, DocumentWorkerDocumentTask.class);
+        Assert.assertTrue(taskData.customData.containsKey(WorkflowWorkerConstants.CustomData.PROJECT_ID),
+                "Custom data on retrieved task data should have a project ID.");
+        Assert.assertEquals(taskData.customData.get(WorkflowWorkerConstants.CustomData.PROJECT_ID), projectId,
+                "Project ID on task should match that specified in properties.");
+        Assert.assertTrue(taskData.customData.containsKey(WorkflowWorkerConstants.CustomData.WORKFLOW_ID),
+                "Custom data on retrieved task data should have a workflow ID.");
+        Assert.assertEquals(taskData.customData.get(WorkflowWorkerConstants.CustomData.WORKFLOW_ID),
+                Long.toString(workflowId), "Workflow ID on task should match that specified in properties.");
+        Assert.assertTrue(taskData.customData.containsKey(WorkflowWorkerConstants.CustomData.OUTPUT_PARTIAL_REFERENCE),
+                "Custom data on retrieved task data should have an output partial reference.");
+        Assert.assertEquals(taskData.customData.get(WorkflowWorkerConstants.CustomData.OUTPUT_PARTIAL_REFERENCE),
+                expectedDataDir,
                 "Output Partial Reference should match the data directory specified in storage properties.");
-        Document document = taskData.getDocument();
+        DocumentWorkerDocument document = taskData.document;
         Assert.assertNotNull(document, "Document on task data should not be null.");
-        Assert.assertTrue(document.getMetadata().containsKey("storageReference"),
+        Assert.assertTrue(document.fields.containsKey("storageReference"),
                 "Storage reference field should be on document.");
-        String storageReferenceValue = document.getMetadata().get("storageReference").iterator().next();
-        Assert.assertNotNull(storageReferenceValue, "Storage reference field should not be null.");
 
-        Assert.assertFalse(encounteredStorageReferences.contains(storageReferenceValue),
-                "Storage reference on the document should not match any other document encountered so from the queue.");
-        encounteredStorageReferences.add(storageReferenceValue);
+        DocumentWorkerFieldValue storageReferenceValue = document.fields.get("storageReference").iterator().next();
+        Assert.assertNotNull(storageReferenceValue, "Storage reference field should not be null.");
+        Assert.assertEquals(storageReferenceValue.encoding, DocumentWorkerFieldEncoding.utf8,
```

Comment:
As discussed I think that this should have an encoding of storage _ref

Reply:
Updated. Requires change from index woker also.